fastjet/internal/predicates/orientation.go

```diff
@@ -91,3 +104,26 @@ func robustOrientation(x1, y1, x2, y2, x, y *big.Rat) OrientationKind {
 		return Colinear
 	}
 }
+
+func matOrientation(x1, y1, x2, y2, x, y float64) OrientationKind {
+	if (x1 == x2 && x2 == x) || (y1 == y2 && y2 == y) {
+		// points are horizontally or vertically aligned
+		return Colinear
+	}
+	m := mat.NewDense(3, 3, []float64{x1, y1, 1, x2, y2, 1, x, y, 1})
+	logDet, sign := mat.LogDet(m)
+	// FIXME determine maxErrorLog
+	// maxErrorLog is the highest possible number that could lead to a wrong sign due to rounding error
+	maxErrorLog := -10.0
+	if logDet < maxErrorLog {
```

Comment:
That's what I had originally. I'm sorry, I forgot to mention that I found a couple cases, all with higher numbers, where mat returns the wrong result. For example for the test
`{1000,2000, 2000, 3000, 10000, 11000, IndeterminateO, Colinear},`
mat returns Counterclockwise.